wagtailio/project_styleguide/templates/patterns/pages/blog/blog_index_page.html

```diff
@@ -9,8 +9,55 @@
+                        |              |  p1   |  p1   |  p1   |  p1   |  p1   |
+                        | 2            |  p2   |  p2   |  p2   |  p2   |  p2   |
+                        | 3            | FP2   | FP2   |       |       |       |
+                        |              |  p3   |  p3   |  p3   |  p3   |  p3   |
+                        | 4            |  p4   |  p4   |  p4   |  p4   |  p4   |
+                        | 5            | FP3   | FP3   | FP2   | FP2   |       |
+                        |              |  p5   |  p5   |  p5   |  p5   |  p5   |
+                        | 6            |  p6   |  p6   |  p6   |  p6   |  p6   |
+                        | 7            | FP4   | FP4   |       |       |       |
+                        |              |  p7   |  p7   |  p7   |  p7   |  p7   |
+                        | 8            |  p8   |  p8   |  p8   |  p8   |  p8   |
+                        | 9            | FP5   |       | FP3   |       |       |
+                        |              |  p9   |  p9   |  p9   |  p9   |  p9   |
+                        | 10           |  p10  |  p10  |  p10  |  p10  |  p10  |
+
+                    {% endcomment %}
+                    {% if forloop.counter|divisibleby:2 %}
+                        {% include 'patterns/pages/blog/includes/blog_post.html' with post=post %}
+                    {% else %}
+                        {% if forloop.first %}
+                            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.0 featured=True %}
+                        {% elif forloop.counter == 3 and featured_posts|length >= 4 %}
+                            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.1 featured=True %}
+                        {% elif forloop.counter == 5 and featured_posts|length >= 4 %}
+                            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.2 featured=True %}
+                        {% elif forloop.counter == 5 and featured_posts|length == 2 or forloop.counter == 5 and featured_posts|length == 3 %}
+                            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.1 featured=True %}
+                        {% elif forloop.counter == 7 and featured_posts|length >= 4 %}
+                            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.3 featured=True %}
+                        {% elif forloop.counter == 9 and featured_posts|length == 5 %}
+                            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.4 featured=True %}
+                        {% elif forloop.counter == 9 and featured_posts|length == 3 %}
+                            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.2 featured=True %}
+                        {% endif %}
```

Comment:
How about something like:

```django+html
{% with num_featured_posts=featured_posts|length %}
    {% if forloop.first %}
        {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.0 featured=True %}
    {% elif forloop.counter == 3 and num_featured_posts >= 4 %}
        {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.1 featured=True %}
    {% elif forloop.counter == 5 %}
        {% if num_featured_posts >= 4 %}
            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.2 featured=True %}
        {% elif num_featured_posts == 2 or num_featured_posts == 3 %}
            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.1 featured=True %}
        {% endif %}
    {% elif forloop.counter == 7 and num_featured_posts >= 4 %}
        {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.3 featured=True %}
    {% elif forloop.counter == 9 %} 
        {% if num_featured_posts == 5 %}
            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.4 featured=True %}
        {% elif num_featured_posts == 3 %}
            {% include 'patterns/pages/blog/includes/blog_post.html' with post=featured_posts.2 featured=True %}
        {% endif %}
    {% endif %}
{% endwith %}
```

Note: not tested locally.


Other than that, let's merge it/

Reply:
Thanks, tested locally and working well!